snitun/server/worker.py

```diff
@@ -62,8 +69,48 @@ async def _async_init(self) -> None:
             throttling=self._throttling,
             event_callback=self._event_stream,
         )
+
+        # Initialize metrics collector in child process
+        self._metrics = self._metrics_factory()
+
         self._list_sni = SNIProxy(self._peers)
-        self._list_peer = PeerListener(self._peers)
+        self._list_peer = PeerListener(self._peers, metrics=self._metrics)
+
+        # Start metrics reporting task
+        self._metrics_task = asyncio.create_task(self._report_metrics_loop())
+
+    async def _report_metrics_loop(self) -> None:
+        """Schedule periodic metrics reporting."""
+        try:
+            while True:
+                await asyncio.sleep(self._metrics_interval)
+                await self._collect_and_report_metrics()
+        except asyncio.CancelledError:
+            # Task was cancelled during shutdown
+            pass
+
+    async def _collect_and_report_metrics(self) -> None:
+        """Collect current state and report metrics."""
+        if not self._metrics or not self._peers:
+            return
+
+        # Count peers by protocol version
+        protocol_version_counts: dict[int, int] = {}
+        total_peers = 0
+
+        for peer in self._peers.iter_peers():
+            total_peers += 1
```

Comment:
As we already have the total (`sum(protocol_version_counts.values()))`), why do we need to keep track of the total in memory here as well?